hypervisor/debug/vuart.c

```diff
@@ -133,13 +133,9 @@ static void vuart_toggle_intr(struct acrn_vuart *vu)
 	intr_reason = vuart_intr_reason(vu);
 
 	if (intr_reason != IIR_NOPEND) {
-		vpic_set_irq(vu->vm, COM1_IRQ, GSI_SET_HIGH);
-
-		vioapic_set_irq(vu->vm, COM1_IRQ, GSI_SET_HIGH);
-
-		vpic_set_irq(vu->vm, COM1_IRQ, GSI_SET_LOW);
-
-		vioapic_set_irq(vu->vm, COM1_IRQ, GSI_SET_LOW);
+		/* active low for COM1_IRQ polarity */
+		vpic_set_irq(vu->vm, COM1_IRQ, GSI_FALLING_PULSE);
+		vioapic_set_irq(vu->vm, COM1_IRQ, GSI_FALLING_PULSE);
```

Comment:
http://www.ti.com/lit/ds/symlink/pc16550d.pdf   for the specification.
The INTR is active high.  

If we still have issue in this patch, please remove this one for now.  We can revisit later on.

Reply:
static void vuart_toggle_intr(struct acrn_vuart *vu)
{
        uint8_t intr_reason;

        intr_reason = vuart_intr_reason(vu);

        if (intr_reason != IIR_NOPEND) {
                vpic_set_irq(vu->vm, COM1_IRQ, GSI_SET_HIGH);

                vioapic_set_irq(vu->vm, COM1_IRQ, GSI_SET_HIGH);

                vpic_set_irq(vu->vm, COM1_IRQ, GSI_SET_LOW);

                vioapic_set_irq(vu->vm, COM1_IRQ, GSI_SET_LOW);
        }
	=
		If (  //has IRQ ) {
			SET_HIGH
		}
		Else
			SET_LOW.
}

Reply:
SOS guest actually set vIOAPIC to low active, if there is no such change, the series console can not work..